sensor/FlowCompressionSensor.js

```diff
@@ -35,12 +37,30 @@ class FlowCompressionSensor extends Sensor {
   }
 
   async run() {
-    setTimeout(async () => {
+    sem.once('IPTABLES_READY', async () => {
       await this.build()
-    }, 10 * 1000) // first time in 10s
-    setInterval(async () => {
-      await this.build()
-    }, 30 * 60 * 1000);
+      setInterval(async () => {
+        await this.build()
+      }, 30 * 60 * 1000);
+    })
+  }
+
+  async checkAndCleanMem() {
+    let compressedMem = 0
+    let oldestKey
+    const compressedFlowsKeys = await rclient.scanResults(this.getKey("*", "*"), 1000)
+    for (const key of compressedFlowsKeys) {
+      const mem = Number(await rclient.memoryAsync("usage", key) || 0)
+      compressedMem += mem
+      if (!oldestKey) {
+        oldestKey = key
+      } else if (await rclient.ttlAsync(key) < await rclient.ttlAsync(oldestKey)) {
```

Comment:
If comparing begin and end in the key name is sufficient, don't invoke 2 redis calls here